File: mine-mart-fe/src/types/customer.ts
/**
 * Customer types - maps to backend DTOs
 */

export interface CustomerResponse {
  id: number;
  fullName: string;
  phone?: string;
  email?: string;
  address?: string;
  note?: string;
}

export interface CustomerCreateRequest {
  fullName: string;
  phone?: string;
  email?: string;
  address?: string;
  note?: string;
}

export interface CustomerUpdateRequest {
  fullName?: string;
  phone?: string;
  email?: string;
  address?: string;
  note?: string;
}

File: mini-mart/src/main/java/com/example/mini_mart/entity/Customer.java
package com.example.mini_mart.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "customer")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 250)
    @NotNull
    @Column(name = "full_name", nullable = false, length = 250)
    private String fullName;

    @Size(max = 50)
    @Column(name = "phone", length = 50)
    private String phone;

    @Size(max = 255)
    @Column(name = "email")
    private String email;

    @Size(max = 1000)
    @Column(name = "address", length = 1000)
    private String address;

    @Size(max = 1000)
    @Column(name = "note", length = 1000)
    private String note;

    @OneToMany(mappedBy = "customer")
    private List<ExportReceipt> exportReceipts = new ArrayList<>();

}